File: Coding Challenge/Exam Challenge/worker Java Files/Student.java
package pattern;

import pattern.worker.Worker;

public class Student extends Worker{

	@Override
	public void work() {
		System.out.println("My Work is to get projects and assignment done on time ");
		w.work();
	}

	@Override
	public void nextWorker(Worker worker) {
		w = worker;
	}


}

File: Coding Challenge/Program 7/Student.cpp
#include "Student.h"
#include "Worker.h"
#include<string>
using namespace pattern;
using namespace worker;

Student::Student(){

}

void Stuent::Worker(){
  cout<<"my work is to get project and assignment done on time"<<endl;
  w.worker();
}

  void Student::nextWorker(Worker worker){
      w = worker;

  }

Student::~Student(){

}
